packages/components/src/public/QueryModel/GridFilterModal.tsx

```diff
@@ -19,12 +19,12 @@ interface Props {
     model: QueryModel;
     onApply: (filters: Filter.IFilter[]) => void;
     onCancel: () => void;
-    selectDistinctOptions?: Query.SelectDistinctOptions;
+    getSelectDistinctOptions?: (column: string, allFilters?: boolean) => Query.SelectDistinctOptions;
```

Comment:
In general, retrieving props from a function is an anti-pattern. The reason for this is the flow of data is going in the wrong direction as something could change upstream that would not be recognized by React which puts the components in possibly intractable state.

In this case I might recommend handing in the different sets of filters as different props.

Reply:
Good point. Updated to go back to passing in `selectDistinctOptions`

Reply:
Thanks for working through that. Go M's.